index.js

```diff
@@ -1,26 +1,118 @@
 // 1.Fibonacci
 
-// function fibo(n) {
-//   return n <= 1 ? n : fibo(n - 1) + fibo(n - 2);
-// }
-// fibo(5);
-// fibo(6);
-// fibo(7);
+function fibo(n) {
+  return n <= 1 ? n : fibo(n - 1) + fibo(n - 2);
+}
+fibo(5);
+fibo(6);
+fibo(7);
 
 // 2. Implementing with reduce
 
 //concat;
+let arr = [1, 2];
+let arr1 = [3, 4];
+
+const concatFn = (array, newEl) =>
+  newEl.reduce((array, newEl) => (array.push(newEl), array), array);
+
+concatFn(arr, arr1);
+
+//filter and map;
+
+let friends = [
+  {
+    name: "John",
+    age: 15,
+  },
+  {
+    name: "Michael",
+    age: 25,
+  },
+  {
+    name: "Lisa",
+    age: 30,
+  },
+];
+
+let adult = friends.reduce(function (newArr, person) {
```

Comment:
please don't mutate state, use spread instead of push

Reply:
but here is a new array -- i didnt mutate it